exporter/prometheusexporter/accumulator.go

```diff
@@ -327,3 +347,53 @@ func copyMetricMetadata(metric pmetric.Metric) pmetric.Metric {
 
 	return m
 }
+
+func accumulateHistogramValues(prev, current, dest pmetric.HistogramDataPoint) {
+	dest.SetStartTimestamp(prev.StartTimestamp())
+
+	older := prev
+	newer := current
+	if current.Timestamp().AsTime().Before(prev.Timestamp().AsTime()) {
+		older = current
+		newer = prev
+	}
+
+	newer.Attributes().CopyTo(dest.Attributes())
+	dest.SetTimestamp(newer.Timestamp())
+
+	// checking for bucket boundary alignment, optionally re-aggregate on newer boundaries
+	match := true
+	if older.ExplicitBounds().Len() == newer.ExplicitBounds().Len() {
+		for i := 0; i < newer.ExplicitBounds().Len(); i++ {
+			if older.ExplicitBounds().At(i) != newer.ExplicitBounds().At(i) {
+				match = false
+				break
+			}
+		}
+	} else {
+		match = false
+	}
+
+	if match {
+		dest.SetCount(newer.Count() + older.Count())
+		if newer.HasSum() && older.HasSum() {
```

Comment:
I added this check for disagreement 
```
else if ip.HasSum() != pp.HasSum() {
    a.logger.With(
	    zap.String("metric_name", metric.Name()),
    ).Warn("Dropped histogram data point due to disagreement on sum tracking")
    continue
}
``` 
However, I could use more of your help here, as I am not sure if I'm doing it right(I also didn't write the original block of code). 

For one, when does it make for a histogram metric to come without sum? In other words, can we simply assume they have sums always? 

Also, assuming it makes sense for a sum to be absent(and thus the necessity to check), should we also be checking for presence of a min and max then?

Reply:
Speaking of min and max, I just realized the code as written right now will not record any min and max values. That would be a bug correct?

Reply:
The Prometheus metrics data model does not include min/max for histograms as all components of a Prometheus histogram are actually counter series and counters are cumulative monotonic sums.  It wouldn't be possible to track a lower `min` value once the first minimum was set and it doesn't generally make sense to track min/max over the entire lifetime of a series.  Instead it makes more sense to be able to track the min/max observed during any given window, such as is possible with a delta histogram.

From the [OTel metrics spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#histogram):

> The aggregation temporality also has implications on the min and max fields. Min and max are more useful for Delta temporality, since the values represented by Cumulative min and max will stabilize as more events are recorded. Additionally, it is possible to convert min and max from Delta to Cumulative, but not from Cumulative to Delta. When converting from Cumulative to Delta, min and max can be dropped, or captured in an alternative representation such as a gauge.

Reply:
Got it. What do you think about that `HasSum` check? The OTEL spec you linked states that sum is actually a required field. In that case, I would think it'd be safe to remove the check and make the code look much nicer

Reply:
Hey Anthony, I went ahead and remove the `HasSum` check. The reason was what I stated above, plus the fact that, base on my research, we can generally assume good data points in exporter as receiver should have handled many of the bad cases. Let me know if that makes sense. Thanks!